TACC/Core-Styles
# feat: new client RT (Request Tracker)

## Overview

Create styles to modify RT `elevator-light.css`.

## Status

> [!TIP]
> This is actively used on https://consult.tacc.utexas.edu/.

> [!IMPORTANT]
> Pending decision for [Core-Styles v3](https://tacc-main.atlassian.net/wiki/x/V4BgAg) of how best to skin vendor CSS.

## Related

- [CMD-75](https://tacc-main.atlassian.net/browse/CMD-75)

## Changes

- **added** `core-styles.rt.elevator-light.css`

## Testing

1. Copy [CDN URL to `core-styles.tr.elevator-light.css` at this branch](https://cdn.jsdelivr.net/gh/TACC/core-styles@feat/new-client-rt/src/lib/_imports/core-styles.rt.elevator-light.css).
2. Open https://consult-dev.tacc.utexas.edu/Admin/Tools/Theme.html.
3. Import the CSS into "Custom CSS (Advanced)" of the [RT Theme Editor](https://docs.bestpractical.com/rt/5.0.5/customizing/styling_rt.html#RT-Theme-Editor) e.g. `@import url("…");`
4. Save changes.
5. Verify header has TACC logo and colors.
6. Verify buttons have TACC button styles.
7. Verify no UX and UI is broken.

## UI

<details open><summary><strong>Round 2 - Min. Effort</strong> (from <a href="https://github.com/TACC/Core-Styles/pull/300/commits/f08ac37d559f79b5cefddc4d052dd69555289e55">f08ac37</a>)</summary>

| light mode | dark mode (no CSS change) |
| - | - |
| <img width="960" alt="rt light mode - header + buttons" src="https://github.com/TACC/Core-Styles/assets/62723358/f1c1a0bd-ca73-4d79-9be7-8a29e79d5531"> | <img width="960" alt="rt dark mode - no CSS change" src="https://github.com/TACC/Core-Styles/assets/62723358/c29e3160-0917-4eb5-a07f-c66d426aa1b1"> |

</details>

<details><summary><strong>Round 1 - Max. Effort</strong> (from <a href="https://github.com/TACC/Core-Styles/pull/300/commits/98ac7e6bbd4af546a842354fb372f7ce34334628">98ac7e6</a>)</summary>

> **Warning** This lacks button styles from Round 2 (f08ac37)

<img width="960" alt="Screenshot 2024-01-17 at 6 14 48 PM" src="https://github.com/TACC/Core-Styles/assets/62723358/76ed8a29-1bcf-4614-830f-2b5fa29ad28a">

</details>
